src/Compilers/Test/Utilities/CSharp/TestSources.cs

```diff
@@ -208,47 +208,164 @@ public static T[] ToArray<T>(void* ptr, int count)
 }";
 
         internal const string Index = @"
+
 namespace System
 {
-    public readonly struct Index
+    using System.Runtime.CompilerServices;
+    public readonly struct Index : IEquatable<Index>
```

Comment:
>Index [](start = 27, length = 5)

I assume this is copied from corefx, so doesn't need reviewing. Let me know if otherwise.